src/RPackage-Core/RPackageOrganizer.class.st

```diff
@@ -232,6 +230,13 @@ RPackageOrganizer >> defineUnpackagedClassesPackage [
 { #category : 'registration' }
 RPackageOrganizer >> ensurePackage: aPackage [
 
+	| packageName |
+	packageName := aPackage isString
+		               ifTrue: [ aPackage ]
+		               ifFalse: [ aPackage name ].
+
+	(self hasPackage: aPackage) ifTrue: [ ^ self packageNamed: packageName ].
+
 	^ aPackage isString
 		  ifTrue: [ self packageNamed: aPackage ifAbsent: [ self registerPackage: (RPackage named: aPackage organizer: self) ] ]
 		  ifFalse: [
```

Comment:
Since there is an `hasPackage:` check that returns if true, it's not necessary to check again in the return statement, assuming `packageNamed:ifAbsent:` detects the absence the same way as `hasPackage:`:
```st
aPackage isString
		  ifTrue: [ self registerPackage: (RPackage named: aPackage organizer: self) ]
		  ifFalse: [
			  self registerPackage: aPackage.
			  aPackage ]
```

Reply:
Indeed! I'll update that in a future PR :)